### DISCUSSION TITLE
Couple of Errors

### DISCUSSION THREAD

**Discussion by donghquinn** [docker-library/ghost#344] [General]
I found couple of Errors when I initiated the ghost with docker.
I have used all latest, 5.22.9-alpine & 5-alpine image versions but nothing different

First, I Found 'Could Not understand Request' Error when I access "/ghost" at the very first time right after started the ghost container.

Here's the Log below

>ghost  | [2022-11-10 08:27:57] ERROR "GET /ghost/api/admin/posts/?formats=mobiledoc%2Clexical&limit=5&filter=status%3A%5Bpublished%2Csent%5D&order=published_at%20desc" 400 41ms
ghost  |
ghost  | Could not understand request.
ghost  |
ghost  | Error ID:
ghost  |     9466a660-60d1-11ed-9417-c319eca350fd
ghost  |
ghost  | Error Code:
ghost  |     ER_BAD_FIELD_ERROR
ghost  |
ghost  | ----------------------------------------
ghost  |
ghost  | Error: select `posts`.*, (with `k` as (select `member_id` from `members_subscription_created_events` where posts.id = members_subscription_created_events.attribution_id union select `member_id` from `members_created_events` where posts.id = members_created_events.attribution_id) select count(*) from `k`) as `count__conversions`, `posts`.*, (select count(distinct `members_click_events`.`member_id`) from `members_click_events` inner join `redirects` on `members_click_events`.`redirect_id` = `redirects`.`id` where posts.id = redirects.post_id) as `count__clicks`, `posts`.*, (select COALESCE(ROUND(AVG(score) * 100), 0) from `members_feedback` where posts.id = members_feedback.post_id) as `count__sentiment`, `posts`.*, (select count(*) from `members_feedback` where posts.id = members_feedback.post_id AND members_feedback.score = 0) as `count__negative_feedback`, `posts`.*, (select sum(`score`) from `members_feedback` where posts.id = members_feedback.post_id) as `count__positive_feedback` from `posts` where (`posts`.`status` in ('published', 'sent') and `posts`.`type` = 'post') order by `posts`.`published_at` DESC limit 5 - Unknown column 'posts.id' in 'where clause'
ghost  |     at /var/lib/ghost/versions/5.22.9/node_modules/@tryghost/bookshelf-pagination/lib/bookshelf-pagination.js:259:27
ghost  |     at Packet.asError (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/packets/packet.js:728:17)
ghost  |     at Query.execute (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/commands/command.js:29:26)
ghost  |     at Connection.handlePacket (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/connection.js:456:32)
ghost  |     at PacketParser.onPacket (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/connection.js:85:12)
ghost  |     at PacketParser.executeStart (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/packet_parser.js:75:16)
ghost  |     at Socket.<anonymous> (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/connection.js:92:25)
ghost  |     at Socket.emit (node:events:513:28)
ghost  |     at addChunk (node:internal/streams/readable:315:12)
ghost  |     at readableAddChunk (node:internal/streams/readable:289:9)
ghost  |     at Socket.Readable.push (node:internal/streams/readable:228:10)
ghost  |     at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
ghost  |

Second, I found Something Fatal Errors When I went into Post menu.
Please Look at the attached Image.

<img width="1416" alt="Screen Shot 2022-11-10 at 5 28 12 PM" src="https://user-images.githubusercontent.com/98877622/201040579-65049f3d-5197-4747-90a2-3187d869efe1.png">

And The Log is below

> ghost  | [2022-11-10 08:28:01] ERROR "GET /ghost/api/admin/posts/?formats=mobiledoc%2Clexical&limit=30&page=1&filter=status%3A%5Bdraft%2Cscheduled%2Cpublished%2Csent%5D" 400 25ms
ghost  |
ghost  | Could not understand request.
ghost  |
ghost  | Error ID:
ghost  |     96d3fce0-60d1-11ed-9417-c319eca350fd
ghost  |
ghost  | Error Code:
ghost  |     ER_BAD_FIELD_ERROR
ghost  |
ghost  | ----------------------------------------
ghost  |
ghost  | Error: select `posts`.*, (with `k` as (select `member_id` from `members_subscription_created_events` where posts.id = members_subscription_created_events.attribution_id union select `member_id` from `members_created_events` where posts.id = members_created_events.attribution_id) select count(*) from `k`) as `count__conversions`, `posts`.*, (select count(distinct `members_click_events`.`member_id`) from `members_click_events` inner join `redirects` on `members_click_events`.`redirect_id` = `redirects`.`id` where posts.id = redirects.post_id) as `count__clicks`, `posts`.*, (select COALESCE(ROUND(AVG(score) * 100), 0) from `members_feedback` where posts.id = members_feedback.post_id) as `count__sentiment`, `posts`.*, (select count(*) from `members_feedback` where posts.id = members_feedback.post_id AND members_feedback.score = 0) as `count__negative_feedback`, `posts`.*, (select sum(`score`) from `members_feedback` where posts.id = members_feedback.post_id) as `count__positive_feedback` from `posts` where (`posts`.`status` in ('draft', 'scheduled', 'published', 'sent') and `posts`.`type` = 'post') order by CASE WHEN posts.status = 'scheduled' THEN 1 WHEN posts.status = 'draft' THEN 2 ELSE 3 END ASC,CASE WHEN posts.status != 'draft' THEN posts.published_at END DESC,posts.updated_at DESC,posts.id DESC limit 30 - Unknown column 'posts.id' in 'where clause'
ghost  |     at /var/lib/ghost/versions/5.22.9/node_modules/@tryghost/bookshelf-pagination/lib/bookshelf-pagination.js:259:27
ghost  |     at Packet.asError (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/packets/packet.js:728:17)
ghost  |     at Query.execute (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/commands/command.js:29:26)
ghost  |     at Connection.handlePacket (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/connection.js:456:32)
ghost  |     at PacketParser.onPacket (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/connection.js:85:12)
ghost  |     at PacketParser.executeStart (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/packet_parser.js:75:16)
ghost  |     at Socket.<anonymous> (/var/lib/ghost/versions/5.22.9/node_modules/mysql2/lib/connection.js:92:25)
ghost  |     at Socket.emit (node:events:513:28)
ghost  |     at addChunk (node:internal/streams/readable:315:12)
ghost  |     at readableAddChunk (node:internal/streams/readable:289:9)
ghost  |     at Socket.Readable.push (node:internal/streams/readable:228:10)
ghost  |     at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
ghost  |

So I can't Do Any Stuffs because they don't understand my request.
But Database migration has no issues.

**Comment by wglambert** [comment]
Are you using MariaDB or Mysql:5.*? https://github.com/TryGhost/Ghost/issues/15729
https://github.com/TryGhost/Ghost/issues/15766

**Comment by donghquinn** [comment]
> Are you using MariaDB or Mysql:5.*? [TryGhost/Ghost#15729](https://github.com/TryGhost/Ghost/issues/15729) [TryGhost/Ghost#15766](https://github.com/TryGhost/Ghost/issues/15766)

I'm using Mariadb

**Comment by donghquinn** [comment]
I've deployed my blog with 4.48.9 version just right after I wrote this Discussion(It was Issue at that time anyway). And today, I challenged to upgrade it  to 5.24.2 version and Successfully did it.

Okay Here's My conclusion: There are something Errors initiating with version 5 or more version at the very first time. However It would be okay to initiate ghost with version 4.x.x and then upgrade it to version 5 or more somehow.

And it means it will be okay if you were already using ghost. If you are first to deploy ghost, then start with 4.x.x version.

And I guess somewhere with data initiating into DB might has Logical Issues according to the errors that I caught above. I hope you check.

Any Idea or opinion?

Thank you
